h5grove/encoders.py

```diff
@@ -99,32 +111,42 @@ def encode(content, encoding: Optional[str] = "json") -> Response:
     :param content: Content to encode
     :param encoding:
         - `json` (default)
-        - `npy`: nD array-like in npy files
         - `bin`: nD array/scalars in bytes
-        - `tiff`: 2D arrays as a TIFF file
+        - `csv`: nD arrays in downloadable csv files
+        - `npy`: nD arrays in downloadable npy files
+        - `tiff`: 2D arrays in downloadable TIFF files
     :returns: A Response object containing content and headers
-    :raises ValueError: If encoding is not "json" nor "npy"
+    :raises ValueError: If encoding is not among the ones above.
     """
     if encoding in ("json", None):
         return Response(
             orjson_encode(content),
             headers={"Content-Type": "application/json"},
         )
 
-    if encoding == "npy":
+    if encoding == "bin":
         return Response(
-            npy_stream(content),
+            bin_encode(content),
             headers={
                 "Content-Type": "application/octet-stream",
-                "Content-Disposition": 'attachment; filename="data.npy"',
             },
         )
 
-    if encoding == "bin":
+    if encoding == "csv":
         return Response(
-            bin_encode(content),
+            convert_csv(content),
+            headers={
+                "Content-Type": "text/csv",
+                "Content-Disposition": 'attachment; filename="data.csv"',
+            },
+        )
+
+    if encoding == "npy":
+        return Response(
+            convert_npy(content),
```

Comment:
The diff is awful because I reorganized things. 

I added the `csv` export and I changed the behaviour of `npy` by replacing `npy_stream` by `convert_npy`. 

That is to say, it does not stream anymore the npy file but send it as an attachment. I don't think the NPY streaming feature was used anywhere so I think this is fine (?)

Reply:
That's fine for me. I did it to give a try to streaming. From what I remember it was only with flask.
You can simplify `Response`'s `content` typing then... and probably the code using it.

Also if we don't care about compatibility (should we as a 0.0.x?), I'd remove `npy_stream` if it is no longer used.

Reply:
> You can simplify Response's content typing then... and probably the code using it.

I would agree but I want to try to improve the performance by using chunks so I will keep like this for now if you don't mind.

Reply:
No problem!